NeoML/include/NeoML/Dnn/Layers/MatrixMultiplicationLayer.h

```diff
@@ -41,6 +44,17 @@ class NEOML_API CMatrixMultiplicationLayer : public CBaseLayer {
 	void RunOnce() override;
 	void BackwardOnce() override;
 	int BlobsForBackward() const override { return TInputBlobs; }
+
+private:
+	enum TSMMD { TSMMD_Forward, TSMMD_Backward, TSMMD_SecondBackward, /*...*/ TSMMD_Count_ };
+	const CSmallMatricesMultiplyDesc* initSmallMatricesMulDesc( TSMMD, int firstHeight, int firstWidth, int secondWidth, int resultWidth );
+	void destroySmallMatricesMulDesc();
+
+	struct CPair final {
```

Comment:
From the amount of copy-paste it looks like you need to do smth with this structure.
I got the idea: the CPU x86 desc itself becomes this structure. (array becomes an array of this structure).
I mean all non-MKL initMatMulDesc returns `nullptr`.
MKL initMatMulDesc never returns `nullptr`. It returns pointer to this structure. And MatMul methods inside check
```
if( desc != nullptr && desc->jit != nullptr ) {
    mkl_call_jit( desc->jit, ... );
} else {
    mkl_nojit_function
}
```

As a result:
* no copy-pasted CPair
* the 2-dimensional array in fully-connected becomes CArray<CPointerArray>
* it looks like that you don't even need explicit `Fault` flag in this case

Reply:
fixed